src/providers/wfs/qgsoapifcollection.cpp

```diff
@@ -0,0 +1,497 @@
+              else if ( values.size() == 6 ) // with zmin at [2] and zmax at [5]
+              {
+                if ( firstBbox )
+                {
+                  mBbox.set( values[0], values[1], values[3], values[4] );
+                }
+                spatialExtent.bounds = QgsBox3d( values[0], values[1], values[2],
+                                                 values[3], values[4], values[5] );
+              }
+              if ( values.size() == 4 || values.size() == 6 )
+              {
+                spatialExtents << spatialExtent;
+                firstBbox = false;
+              }
+            }
+          }
+          metadataExtent.setSpatialExtents( spatialExtents );
+        }
+      }
+    }
+#ifndef REMOVE_SUPPORT_DRAFT_VERSIONS
+    else if ( extent.is_object() && extent.contains( "bbox" ) )
+    {
+      const auto bbox = extent["bbox"];
+      if ( bbox.is_array() && bbox.size() == 4 )
+      {
+        std::vector<double> values;
+        for ( size_t i = 0; i < bbox.size(); i++ )
+        {
+          if ( !bbox[i].is_number() )
+          {
+            values.clear();
+            break;
+          }
+          values.push_back( bbox[i].get<double>() );
+        }
+        if ( values.size() == 4 )
+        {
+          mBbox.set( values[0], values[1], values[2], values[3] );
+          QgsLayerMetadata::SpatialExtent spatialExtent;
+          spatialExtent.extentCrs = QgsCoordinateReferenceSystem::fromOgcWmsCrs(
+                                      QStringLiteral( "http://www.opengis.net/def/crs/OGC/1.3/CRS84" ) );
+          mLayerMetadata.setCrs( spatialExtent.extentCrs );
+          metadataExtent.setSpatialExtents( QList<  QgsLayerMetadata::SpatialExtent >() << spatialExtent );
+        }
+      }
+    }
+#endif
+
+    if ( extent.is_object() && extent.contains( "temporal" ) )
+    {
+      const auto temporal = extent["temporal"];
+      if ( temporal.is_object() && temporal.contains( "interval" ) )
+      {
+        const auto jIntervals = temporal["interval"];
+        if ( jIntervals.is_array() )
+        {
+          QList< QgsDateTimeRange > temporalExtents;
+          for ( const auto &jInterval : jIntervals )
+          {
+            if ( jInterval.is_array() && jInterval.size() == 2 )
+            {
+              QDateTime dt[2];
+              for ( int i = 0; i < 2; i++ )
+              {
+                if ( jInterval[i].is_string() )
+                {
+                  dt[i] = QDateTime::fromString( QString::fromStdString( jInterval[i].get<std::string>() ), Qt::ISODateWithMs );
+                }
+              }
+              if ( !dt[0].isNull() || !dt[1].isNull() )
+              {
+                temporalExtents << QgsDateTimeRange( dt[0], dt[1] );
+              }
+            }
+          }
+          metadataExtent.setTemporalExtents( temporalExtents );
+        }
+      }
+    }
+
+    mLayerMetadata.setExtent( metadataExtent );
+  }
+
+  // From STAC specification
```

Comment:
what is it?

Reply:
A close friend of OAPIF : https://stacspec.org/

Reply:
URL added to comment